pygmt/tests/test_figure.py

```diff
@@ -199,6 +223,27 @@ def test_figure_show_invalid_method():
         fig.show(method="test")
 
 
+@pytest.mark.skipif(IPython is not None, reason="run without IPython installed")
+def test_figure_show_notebook_error_without_ipython():
+    """
+    Test to check if an error is raised when display method is 'notebook', but
+    IPython is not installed.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 1, 2, 3], frame=True)
+    with pytest.raises(GMTError):
+        fig.show(method="notebook")
+
+
+def test_figure_display_external():
+    """
+    Test to check that a figure can be displayed in an external window.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 3, 6, 9], projection="X1c", frame=True)
+    fig.show(method="external")
```

Comment:
The CI can't open external viewers. Why does this test work?

Reply:
Yeah, the console won't show a window pop-up, but I think the code will still execute fine somehow. Looking at the codecov diff at https://app.codecov.io/gh/GenericMappingTools/pygmt/compare/1496/changes#D2L237, the `launch_external_viewer` function is being called up to the `time.sleep(0.5)` line, so it seems that Linux/macOS/Windows launchers (`xdg-open`/`open`/not-sure-what's-on windows) are available on CI?

![image](https://user-images.githubusercontent.com/23487320/133947305-30984ef8-8164-49d1-97d9-ec5321f756a4.png)